### PR TITLE
coding guidelines v2

### PR DESCRIPTION
**Pull request by Jaguar0625**
need to improve (add?) navigation, but content should be mostly done. 

so made review!

### CODE REVIEW

File: source/handbook/all-coding-guidelines.rst [hunk 1]
```diff
@@ -55,25 +55,122 @@ self.assertEqual(['alpha'], words)
...
+For internally consumable projects, it's often enough to just reference the parameters without documenting each in detail.
+
+We believe that code should be for the most part self-documenting so avoid inline comments.
+Of course, there are exceptions for really complicated logic, but those should be exceptions rather than the rule.
+
+Naming
+******
+
+Avoid unpronounceable names and abbreviations.
+Use a spell checker!
```
[the remark this answers is not in the document]
Comment (Jaguar0625) [reply source/handbook/all-coding-guidelines.rst:73]: we do not

**Review by segfaultxavi — CHANGES_REQUESTED** [review]
I've added a commit fixing indentation issues. RST is very picky with indentation, just like Python.
Extra indentation is how you indicate Quote Blocks in RST, so if you don't want these blocks  to appear everywhere, use the right indentation!

File: source/handbook/javascript-coding-guidelines.rst [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+      - tests  # directory containing tests for source code
+        - Parser_spec.js  # unit tests for parser/Parser.js
+        - ast
+          - ast_spec.js  # unit tests for parser/ast/ast.js
+
+      - package.json
+
+Naming
+******
+
```
Comment (0x6861746366574) [inline source/handbook/javascript-coding-guidelines.rst:40]: Should we change to:

1. camelCase should be used for variables, methods and functions. Private variables should be prepended with an underscore. 
2. PascalCase should be used for class and component.
3. TITLE_CASE should be used for constants.

Comment (Jaguar0625) [reply source/handbook/javascript-coding-guidelines.rst:40]: yes, as discussed in Discord this needs updating

File: source/handbook/all-coding-guidelines.rst [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+3. Boolean properties should generally begin with ``is``, ``has`` or ``should``.
+
+   ``isOpen()`` instead of ``open()``
+
+4. Non-boolean properties should generally avoid ``get`` and ``set`` prefixes:
+
+   ``value(12)`` instead of ``setValue(12)``
+
+   ``value()`` instead of ``getValue()``
+
```
Comment (gimre-xymcity) [inline source/handbook/all-coding-guidelines.rst:93]: not boolean, and not sure if applicable to all languages, but what about `try*` functions/methods?

File: source/handbook/cpp-coding-guidelines.rst [hunk 4]
```diff
@@ -1,289 +1,238 @@
...
 
-* In case of operators please put additional space before and after them. This makes code much more readable.
+Prefixes
+--------
 
-  This should be always used in case of ``=``, ``==``, ``!=``, ``&&``, ``||``. So this one's ok:
+1. Prefix class instance members with ``m_``
 
-  .. code-block:: c++
+2. Prefix class static members with ``s_``
```
Comment (gimre-xymcity) [inline source/handbook/cpp-coding-guidelines.rst:74]: hmm, I don't remember do we do this anywhere?

File: source/handbook/cpp-coding-guidelines.rst [hunk 5]
```diff
@@ -1,289 +1,238 @@
...
 
 Special Names
-*************
+-------------
 
 * BlockChain not Blockchain
 
 * Timestamp not TimeStamp
 
 * Filesystem not FileSystem
```
Comment (gimre-xymcity) [inline source/handbook/cpp-coding-guidelines.rst:138]: I think there are more, but we have it in linter and haven't been updating this list...

**Review by gimre-xymcity — DISMISSED** [review]
👏 👍

File: source/handbook/all-coding-guidelines.rst [hunk 6]
```diff
@@ -0,0 +1,186 @@
...
+Logging
+*******
+
+Not all frameworks support the full spectrum of logging levels.
+A description of logging levels from most to least severe follows:
+
+- **FATAL** - Critical error that will require the immediate exit of the program.
+- **ERROR** - Serious error that the node operator might need to act on but will not terminate the program.
+- **WARNING** - Error of which the node operator should be made aware but unlikely requires action.
```
[the remark this answers is not in the document]
Comment (Jaguar0625) [reply source/handbook/all-coding-guidelines.rst:102]: yes, but i think we'd ideally want to change that vs loosening the guidelines, which i view as aspirational :)

File: source/handbook/cpp-coding-guidelines.rst (hunk 4 above)
Comment (Jaguar0625) [reply source/handbook/cpp-coding-guidelines.rst:74]: it's rare, but we have this naming in PacketSocket

File: source/handbook/cpp-coding-guidelines.rst (hunk 5 above)
Comment (0x6861746366574) [reply source/handbook/cpp-coding-guidelines.rst:139]: Please not "BlockChain". 
It is Blockchain. As defined everywhere else in the world. 😅

**Review by gimre-xymcity — APPROVED** [review]

**Pull request merged**